src/visGeometry/index.ts

```diff
@@ -1066,198 +1066,37 @@ class VisGeometry {
         }
     }
 
-    /**
-     * AGENT COLOR HANDLING (TODO: move to separate file)
-     * General notes about data being used to map color to agents:
-     * @property this.colorsData is an array of floats, each 4 floats is a color
-     *  `dataColorIndex` is always an index into the colorsData array, so it is a multiple of 4
-     * `colorId` is always is a number between 0 and numberOfColors-1; ie the index of the color
-     * in the initial colors array.
-     * @property this.idColorMapping uses @param colorId. It maps agent id to colorId
-     *
-     * No other module should know about this.colorsData, or the fact that it's 4 times as
-     * long as the input colors. They should only know about colorId
-     * Therefore all the public color methods should use colorId, and the private methods
-     * can convert between colorId and dataColorIndex
-     */
-
-    /**
-     * Agent color handling: private methods
-     */
-
-    private get numberOfColors(): number {
-        return this.colorsData.length / 4;
-    }
-
     private setAgentColors(): void {
         this.visAgents.forEach((agent) => {
             agent.setColor(
-                this.getColorForTypeId(agent.agentData.type),
-                this.getColorIdForTypeId(agent.agentData.type)
```

Comment:
@toloudis this is the result of changing the interface to `setColor`. Since each of these functions are receiving the same data it made sense to me to make a single function

Reply:
Yup I think the usage pattern makes sense but I would prefer a named type for this bundle of color info.